tests/Unit/NumericalAlgorithms/Interpolation/Test_IrregularInterpolant.cpp

```diff
@@ -0,0 +1,177 @@
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+// IWYU pragma: no_forward_declare MathFunction
+// IWYU pragma: no_forward_declare PowX
+// IWYU pragma: no_forward_declare Tensor
+
+namespace TestTags {
+
+template <size_t Dim>
+struct Vector : db::SimpleTag {
+  using type = tnsr::I<DataVector, Dim>;
+  static std::string name() noexcept {return "Vector"; }
+  static auto fill_values(const MathFunctions::TensorProduct<Dim>& f,
+                          const tnsr::I<DataVector, Dim>& x) noexcept {
+    auto result = make_with_value<tnsr::I<DataVector, Dim>>(x, 0.);
+    const auto f_of_x = f(x);
+    for (size_t d = 0; d < Dim; ++d) {
+      result.get(d) = (d + 0.5) * get(f_of_x);
+    }
+    return result;
+  }
+};
+
+template <size_t Dim>
+struct SymmetricTensor : db::SimpleTag {
+  using type = tnsr::ii<DataVector, Dim>;
+  static std::string name() noexcept {return "SymmetricTensor"; }
+  static auto fill_values(const MathFunctions::TensorProduct<Dim>& f,
+                          const tnsr::I<DataVector, Dim>& x) noexcept {
+    auto result = make_with_value<tnsr::ii<DataVector, Dim>>(x, 0.);
+    const auto f_of_x = f(x);
+    for (size_t i = 0; i < Dim; ++i) {
+      for (size_t j = i; j < Dim; ++j) {  // Symmetry
+        result.get(i, j) = (i + j + 0.33) * get(f_of_x);
+      }
+    }
+    return result;
+  }
+};
+
+}  // namespace TestTags
+
+namespace {
+
+template <size_t Dim>
+void test_interpolate_to_points(const Index<Dim>& mesh) noexcept {
+  // Fill target interpolation coordinates with random values
+  std::random_device r;
+  const auto seed = r();
+  std::mt19937 generator(seed);
+  INFO("seed" << seed);
+  std::uniform_real_distribution<> dist(-1., 1.);
+
+  const auto nn_generator = make_not_null(&generator);
+  const auto nn_dist = make_not_null(&dist);
+
+  const size_t number_of_points = 6;
+  const auto target_x = make_with_random_values<tnsr::I<DataVector, Dim>>(
+      nn_generator, nn_dist, DataVector(number_of_points));
+
+  // Set up interpolator. Need do this only once.
+  const IrregularInterpolant<Dim> irregular_interpolant(mesh, target_x);
+
+  // Coordinates on Legendre-Gauss-Lobatto grid
+  const auto src_x = [&mesh]() {
+    tnsr::I<DataVector, Dim> x(mesh.product());
+    for (size_t d = 0; d < Dim; ++d) {
+      const auto& x_colloc = Basis::lgl::collocation_points(mesh[d]);
+      for (IndexIterator<Dim> iter(mesh); iter; ++iter) {
+        x.get(d)[iter.collapsed_index()] = x_colloc[iter()[d]];
+      }
+    }
+    return x;
+  }();
+
+  // Set up variables
+  using tags =
+      tmpl::list<TestTags::Vector<Dim>, TestTags::SymmetricTensor<Dim>>;
+  Variables<tags> src_vars(mesh.product());
+  Variables<tags> expected_dest_vars(number_of_points);
+
+  // We will make polynomials of the form x^a y^b z^c ...
+  // for all possible a,b,c, so first loop over "a,b,c"
```

Comment:
by "all possible" you mean all 32(64)-bit integers or all integers that would result in exact interpolation?

Reply:
What's the "second" loop? (you say "first loop")